test/test/cpu/CpuTests.java

```diff
@@ -32,4 +43,52 @@ public void itimerTotal(TestProcess p) throws Exception {
         Output out = p.profile("-d 2 -e itimer -i 100ms --total -o collapsed");
         assertCloseTo(out.total(), 2_000_000_000, "itimer total should match profiling duration");
     }
+
+    @Test(mainClass = CpuBurner.class, os = Os.LINUX)
+    public void perfEventsWrongTargetCpu(TestProcess p) throws Exception {
+        pinCpu(p, 1);
+
+        Output out = p.profile("-d 2 -e cpu -i 100ms --total -o collapsed --target-cpu 0");
+        Assert.isEqual(out.total(), 0, "perf_events total should be 0 when the wrong CPU is targeted");
+    }
+
+    @Test(mainClass = CpuBurner.class, os = Os.LINUX)
+    public void perfEventsRightTargetCpu(TestProcess p) throws Exception {
+        pinCpu(p, 0);
+
+        Output out = p.profile("-d 2 -e cpu -i 100ms --total -o collapsed --target-cpu 0");
+        assertCloseTo(out.total(), 2_000_000_000, "perf_events total should match profiling duration");
+    }
+
+    @Test(mainClass = CpuBurner.class, os = Os.LINUX)
+    public void perfEventsWrongTargetCpuWithFdTransfer(TestProcess p) throws Exception {
+        pinCpu(p, 1);
+
+        Output out = p.profile("-d 2 -e cpu -i 100ms --total -o collapsed --target-cpu 0 --fdtransfer");
+        Assert.isEqual(out.total(), 0, "perf_events total should be 0 when the wrong CPU is targeted");
+    }
+
+    @Test(mainClass = CpuBurner.class, os = Os.LINUX)
+    public void perfEventsRightTargetCpuWithFdTransfer(TestProcess p) throws Exception {
+        pinCpu(p, 0);
+
+        Output out = p.profile("-d 2 -e cpu -i 100ms --total -o collapsed --target-cpu 0 --fdtransfer");
+        assertCloseTo(out.total(), 2_000_000_000, "perf_events total should match profiling duration");
+    }
```

Comment:
Same as above for this pair of tests.

Reply:
869b561edfd740e22e67988c617871ef4b2a86a3